### PR TITLE
feat(@aws-amplify/auth): Add endpoint variable in Auth

### PR DESCRIPTION
**Pull request by slashinski**
_Description of changes:_
Add endpoint variable so that it could be used in the Amplify.configure when a user wants to use the cognito-idp-fips endpoint. 
Changes include updating:
**auth**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/src/Client.js [hunk 1]
```diff
@@ -5,9 +5,17 @@ export default class Client {
 	 * Constructs a new AWS Cognito Identity Provider client object
 	 * @param {string} region AWS region
 	 * @param {string} endpoint endpoint
+	 * @param {boolean} fipsEnabled fips flag
 	 */
-	constructor(region, endpoint) {
-		this.endpoint = endpoint || `https://cognito-idp.${region}.amazonaws.com/`;
```
Comment (manueliglesias) [inline packages/amazon-cognito-identity-js/src/Client.js]: If you keep the constructor with no changes, as a consumer, you could just pass your endpoint in the constructor.

File: packages/amazon-cognito-identity-js/index.d.ts [hunk 2]
```diff
@@ -291,6 +291,7 @@ declare module 'amazon-cognito-identity-js' {
 		UserPoolId: string;
 		ClientId: string;
 		endpoint?: string;
+		fipsEnabled?: boolean;
```
Comment (manueliglesias) [inline packages/amazon-cognito-identity-js/index.d.ts]: Not needed if you just override the endpoint

File: packages/amazon-cognito-identity-js/src/CognitoUserPool.js [hunk 3]
```diff
@@ -37,6 +37,7 @@ export default class CognitoUserPool {
 			UserPoolId,
 			ClientId,
 			endpoint,
+			fipsEnabled,
```
Comment (manueliglesias) [inline packages/amazon-cognito-identity-js/src/CognitoUserPool.js]: Not needed if you just override the endpoint

File: packages/amazon-cognito-identity-js/src/Client.js [hunk 4]
```diff
@@ -5,9 +5,17 @@ export default class Client {
...
+	constructor(region, endpoint, fipsEnabled) {
+		if (fipsEnabled) {
+			this.endpoint =
+				endpoint || `https://cognito-idp-fips.${region}.amazonaws.com/`;
+		} else {
+			this.endpoint =
+				endpoint || `https://cognito-idp.${region}.amazonaws.com/`;
+		}
+
```
Comment (manueliglesias) [inline packages/amazon-cognito-identity-js/src/Client.js]: Not needed

File: packages/amazon-cognito-identity-js/src/CognitoUserPool.js [hunk 5]
```diff
@@ -50,7 +51,7 @@ export default class CognitoUserPool {
 		this.userPoolId = UserPoolId;
 		this.clientId = ClientId;
 
-		this.client = new Client(region, endpoint);
+		this.client = new Client(region, endpoint, fipsEnabled);
```
Comment (manueliglesias) [inline packages/amazon-cognito-identity-js/src/CognitoUserPool.js]: Not needed if you just override the endpoint

File: packages/auth/src/Auth.ts [hunk 6]
```diff
@@ -153,6 +153,8 @@ export default class AuthClass {
 			refreshHandlers,
 			identityPoolRegion,
 			clientMetadata,
+			endpoint,
```
Comment (manueliglesias) [inline packages/auth/src/Auth.ts]: This one looks fine

File: packages/auth/src/Auth.ts [hunk 7]
```diff
@@ -153,6 +153,8 @@ export default class AuthClass {
...
+			fipsEnabled,
```
Comment (manueliglesias) [inline packages/auth/src/Auth.ts]: Not needed if you just override the endpoint

File: packages/auth/src/Auth.ts [hunk 8]
```diff
@@ -178,6 +180,8 @@ export default class AuthClass {
 			const userPoolData: ICognitoUserPoolData = {
 				UserPoolId: userPoolId,
 				ClientId: userPoolWebClientId,
+				endpoint,
+				fipsEnabled,
```
Comment (manueliglesias) [inline packages/auth/src/Auth.ts]: same as above

File: packages/auth/src/types/Auth.ts [hunk 9]
```diff
@@ -50,6 +50,8 @@ export interface AuthOptions {
 	authenticationFlowType?: string;
 	identityPoolRegion?: string;
 	clientMetadata?: any;
+	endpoint?: string;
```
Comment (manueliglesias) [inline packages/auth/src/types/Auth.ts]: This one we need

File: packages/auth/src/types/Auth.ts [hunk 10]
```diff
@@ -50,6 +50,8 @@ export interface AuthOptions {
...
+	fipsEnabled?: boolean;
```
Comment (manueliglesias) [inline packages/auth/src/types/Auth.ts]: Not needed if you just override the endpoint

File: packages/amazon-cognito-identity-js/src/Client.js (hunk 1 above)
Comment (slashinski) [reply packages/amazon-cognito-identity-js/src/Client.js]: That is a good point and I totally agree.  I actually went that route internally but saw a request from a while to have a boolean.  I will update accordingly and recommit.

**Comment by slashinski** [comment]
Changes requested have been committed, please review again.

**Comment by ashika01** [comment]
@slashinski I will close this one in favor of #6124. I will add you to list as co-author. Thank you for contribution. Sorry this was not brought in earlier.

**Pull request closed without merging**